experiment_pages/experiment/map_rfid.py

```diff
@@ -46,6 +46,27 @@ def __init__(self, database, parent: CTk, previous_page: CTkFrame = None):
                                       width=15, command=self.add_random_rfid)
         simulate_rfid_button.place(relx=0.80, rely=0.17, anchor=CENTER)
 
+        # New Simulate All RFID Button
+        simulate_all_rfid_button = CTkButton(self, text="Simulate All RFID", compound=TOP,
+                                     width=15, command=self.simulate_all_rfid)
+            simulate_all_rfid_button.place(relx=0.80, rely=0.27, anchor=CENTER)
+
+        def simulate_all_rfid(self):
+        '''Simulates RFID values for all unmapped animals.'''
+        unmapped_count = self.db.get_number_animals() - len(self.animals)
+
+    if unmapped_count <= 0:
```

Comment:
I'm really confused by the indentation here. This can't possibly run, Python likes its indentation.